external-crates/move/crates/move-vm-runtime/src/dev_utils/vm_arguments.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    execution::{
+        TypeSubst as _,
+        dispatch_tables::VMDispatchTables,
+        interpreter::locals::{BaseHeap, BaseHeapId},
+        values::{PrimVec, Struct, Value, Variant, Vector, VectorSpecialization},
+        vm::MoveVM,
+    },
+    jit::execution::ast::Type,
+};
+use move_binary_format::errors::{Location, PartialVMError, PartialVMResult, VMResult};
+use move_core_types::{
+    identifier::IdentStr,
+    language_storage::ModuleId,
+    runtime_value::{MoveStruct, MoveValue, MoveVariant},
+    vm_status::StatusCode,
+};
+use std::{borrow::Borrow, cell::RefCell, collections::BTreeMap, rc::Rc};
+use tracing::warn;
+
+// -------------------------------------------------------------------------------------------------
+// Types
+// -------------------------------------------------------------------------------------------------
+
+/// Serialized return values from function/script execution
+/// Simple struct is designed just to convey meaning behind serialized values
+#[derive(Debug)]
+pub struct ContextualizedValues {
+    pub context: CallContext,
+    /// The value of any arguments that were mutably borrowed.
+    /// Non-mut borrowed values are not included
+    /// Mapping is from argument index to the heap location
+    pub heap_refs: BTreeMap<u16, BaseHeapId>,
+    /// The values passed in with any references taken.
+    pub values: Vec<Value>,
+}
+
+/// Rc and RefCell wrapper around CallContext_ to make it a bit easier and ergonomic to pass around
+/// and mutate for testing purposes.
+pub type CallContext = Rc<RefCell<BaseHeap>>;
```

Comment:
Can we just inline this type? I am a bit confused about why this lives separately.

Reply:
Yea we can..I might have gotten a bit overzealous on the type aliases..